Compiling fails on 32-bit systems

Since #20233 compiling with clang 10.0 fails:
```
bench/addrman.cpp:113:32: error: non-constant-expression cannot be narrowed from type 'unsigned long long' to 'size_t' (aka 'unsigned int') in initializer list [-Wc++11-narrowing]
    const size_t addrman_count{bench.epochs() * bench.epochIterations()};
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bench/addrman.cpp:113:32: note: insert an explicit cast to silence this issue
    const size_t addrman_count{bench.epochs() * bench.epochIterations()};
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                               static_cast<size_t>(                    )
1 error generated.
make[2]: *** [Makefile:12508: bench/bench_bitcoin-addrman.o] Error 1
```

I can reproduce on 64-bit via:

```
$ CONFIG_SITE="$PWD/depends/i686-pc-linux-gnu/share/config.site" ./configure CC="clang -m32" CXX="clang++ -m32"
$ make V=1
Making all in src
make[1]: Entering directory '/bitcoin-core/src'
make[2]: Entering directory '/bitcoin-core/src'
make[3]: Entering directory '/bitcoin-core'
make[3]: Leaving directory '/bitcoin-core'
/usr/bin/ccache clang++ -m32 -std=c++17 -DHAVE_CONFIG_H -I. -I../src/config  -fmacro-prefix-map=/bitcoin-core/src=. -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I. -I./secp256k1/include  -I/bitcoin-core/depends/i686-pc-linux-gnu/include -I./leveldb/include -I./leveldb/helpers/memenv -I./univalue/include -I/bitcoin-core/depends/i686-pc-linux-gnu/include -pthread -I/bitcoin-core/depends/i686-pc-linux-gnu/include -I./bench/ -I/bitcoin-core/depends/i686-pc-linux-gnu/include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_FILE_OFFSET_BITS=64 -DPROVIDE_FUZZ_MAIN_FUNCTION -fdebug-prefix-map=/bitcoin-core/src=. -Wstack-protector -fstack-protector-all -fcf-protection=full      -fPIE -pipe -O2  -MT bench/bench_bitcoin-addrman.o -MD -MP -MF bench/.deps/bench_bitcoin-addrman.Tpo -c -o bench/bench_bitcoin-addrman.o `test -f 'bench/addrman.cpp' || echo './'`bench/addrman.cpp
bench/addrman.cpp:113:32: error: non-constant-expression cannot be narrowed from type 'unsigned long long' to 'size_t' (aka 'unsigned int') in initializer list [-Wc++11-narrowing]
    const size_t addrman_count{bench.epochs() * bench.epochIterations()};
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bench/addrman.cpp:113:32: note: insert an explicit cast to silence this issue
    const size_t addrman_count{bench.epochs() * bench.epochIterations()};
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                               static_cast<size_t>(                    )
1 error generated.

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -235,7 +235,7 @@ task:
     FILE_ENV: "./ci/test/00_setup_env_native_fuzz.sh"
 
 task:
-  name: '[multiprocess, DEBUG] [focal]'
+  name: '[multiprocess, i686, DEBUG] [focal]'
   << : *GLOBAL_TASK_TEMPLATE
   container:
     image: ubuntu:focal
@@ -244,7 +244,7 @@ task:
   env:
     << : *CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
     MAKEJOBS: "-j8"
-    FILE_ENV: "./ci/test/00_setup_env_native_multiprocess.sh"
+    FILE_ENV: "./ci/test/00_setup_env_i686_multiprocess.sh"
 
 task:
   name: '[no wallet] [bionic]'

--- a/ci/test/00_setup_env_i686_multiprocess.sh
+++ b/ci/test/00_setup_env_i686_multiprocess.sh
@@ -6,11 +6,12 @@
 
 export LC_ALL=C.UTF-8
 
-export CONTAINER_NAME=ci_native_multiprocess
+export HOST=i686-pc-linux-gnu
+export CONTAINER_NAME=ci_i686_multiprocess
 export DOCKER_NAME_TAG=ubuntu:20.04
-export PACKAGES="cmake python3 python3-pip llvm clang"
+export PACKAGES="cmake python3 python3-pip llvm clang g++-multilib"
 export DEP_OPTS="DEBUG=1 MULTIPROCESS=1"
 export GOAL="install"
-export BITCOIN_CONFIG="--enable-debug CC=clang CXX=clang++"  # Use clang to avoid OOM
+export BITCOIN_CONFIG="--enable-debug CC='clang -m32' CXX='clang++ -m32' LDFLAGS='--rtlib=compiler-rt -lgcc_s'"
 export TEST_RUNNER_ENV="BITCOIND=bitcoin-node"
 export PIP_PACKAGES="lief"

--- a/src/bench/addrman.cpp
+++ b/src/bench/addrman.cpp
@@ -5,6 +5,7 @@
 #include <addrman.h>
 #include <bench/bench.h>
 #include <random.h>
+#include <util/check.h>
 #include <util/time.h>
 
 #include <optional>
@@ -110,7 +111,8 @@ static void AddrManGood(benchmark::Bench& bench)
      * we want to do the same amount of work in every loop iteration. */
 
     bench.epochs(5).epochIterations(1);
-    const size_t addrman_count{bench.epochs() * bench.epochIterations()};
+    const uint64_t addrman_count{bench.epochs() * bench.epochIterations()};
+    Assert(addrman_count == 5U);
 
     std::vector<std::unique_ptr<CAddrMan>> addrmans(addrman_count);
     for (size_t i{0}; i < addrman_count; ++i) {

